### DISCUSSION TITLE
liquidsoap doesn't play playlists

### DISCUSSION THREAD

**Discussion by suwofis** [savonet/liquidsoap#4110] [User Support]
Hey, liquidsoap works for me now, but somehow I can only stream singles. Playlists just don't work. While starting, the log says the following: ``2024/08/25 20:10:19 >>> LOG START
2024/08/25 20:10:08 [ffmpeg.filter.bitstream:3] No valid mode found for filter pgs_frame_merge!
2024/08/25 20:10:08 [main:3] Liquidsoap 2.1.3
2024/08/25 20:10:08 [main:3] Using: graphics=5.1.2 bytes=[distributed with OCaml 4.02 or above] pcre=7.5.0 sedlex=3.0 menhirLib=20220210 curl=0.9.2 uri=4.2.0 dtools=0.4.4 duppy=0.9.2 cry=0.6.7 mm=0.8.1 xmlplaylist=0.1.5 lastfm=0.3.3 ogg=0.7.3 ogg.decoder=0.7.3 vorbis=0.8.0 vorbis.decoder=0.8.0 opus=0.2.2 opus.decoder=0.2.2 speex=0.4.1 speex.decoder=0.4.1 mad=0.5.2 flac=0.3.1 flac.ogg=0.3.1 flac.decoder=0.3.1 dynlink=[distributed with Ocaml] lame=0.3.6 shine=0.2.3 gstreamer=0.3.1 frei0r=0.1.2 theora=0.4.0 theora.decoder=0.4.0 ffmpeg=1.1.6 bjack=0.1.6 alsa=0.3.0 ao=0.2.4 samplerate=0.1.6 taglib=0.3.10 ssl=0.5.11 magic=0.7.3 camomile=1.0.2 inotify=2.4.1 faad=0.5.1 soundtouch=0.1.9 portaudio=0.2.3 pulseaudio=0.1.5 ladspa=0.2.2 camlimages=4.2.6 camlimages.freetype=5.0.4 lo=0.2.0 gd=1.0a5
2024/08/25 20:10:08 [dynamic.loader:3] Could not find dynamic module for fdkaac encoder.
2024/08/25 20:10:08 [gstreamer.loader:3] Loaded GStreamer 1.22.0 0
2024/08/25 20:10:08 [clock:3] Using builtin (low-precision) implementation for latency control
2024/08/25 20:10:19 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz main.
2024/08/25 20:10:19 [frame:3] Video frame size set to: 1280x720
2024/08/25 20:10:19 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2024/08/25 20:10:19 [frame:3] Targeting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2024/08/25 20:10:19 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2024/08/25 20:10:19 [sandbox:3] Sandboxing disabled
2024/08/25 20:10:19 [video.converter:3] Using preferred video converter: ffmpeg.
2024/08/25 20:10:19 [audio.converter:3] Using samplerate converter: libsamplerate.
2024/08/25 20:10:19 [single_0:3] "/home/user/single.mp3" is static, resolving once for all...
2024/08/25 20:10:19 [single_0:3] Prepared "/home/user/test.mp3" (RID 1).
2024/08/25 20:10:19 [clock.main:3] Streaming loop starts in auto-sync mode
2024/08/25 20:10:19 [clock.main:3] Delegating synchronisation to CPU clock
2024/08/25 20:10:19 [video.text:3] Using camlimages implementation
2024/08/25 20:10:19 [switch_0:3] Switch to single_0.
[mp3float @ 0x7f5c006cc0] Could not update timestamps for skipped samples.
2024/08/25 20:10:20 [Radio_test:3] Connecting mount stream for source@localhost...
2024/08/25 20:10:20 [Radio_test:3] Connection setup was successful.
^C2024/08/25 20:11:07 [main:3] Shutdown started!
2024/08/25 20:11:07 [main:3] Waiting for main threads to terminate...
2024/08/25 20:11:07 [Radio_test:3] Closing connection...
2024/08/25 20:11:07 [clock.main:3] Streaming loop stopped.
2024/08/25 20:11:07 [main:3] Main threads terminated.
2024/08/25 20:11:07 [threads:3] Shutting down scheduler...
2024/08/25 20:11:07 [threads:3] Scheduler shut down.
2024/08/25 20:11:07 [main:3] Cleaning downloaded files...
2024/08/25 20:11:07 [main:3] Freeing memory...

the single.mp3 is the fallback single. the playlist has the name playlist.m3u and contains the path to the songs which are also in the home directory. Rights are correct.
Do you know what the problem is?

**Comment by vitoyucepi** [comment]
Hi @suwofis,
I don't see the error. Could you please paste the code of your script?
It would also be great if you could try installing liquidsoap:2.2.5 from the release artifacts available at https://github.com/savonet/liquidsoap/releases/tag/v2.2.5. The liquidsoap version available in the debian:12 repositories is outdated, so if there is a bug, it won't be fixed for at least a year.

**Reply by suwofis** [reply]
Thanks for your answer and sorry for my late reply. Here is my code:

hauptplaylist = playlist("/home/user/playlist.m3u")


radio = fallback([hauptplaylist])

output.icecast(
        %mp3,
        host = "localhost",
        port = 8000,
        password = "top secret",
        mount = "stream",
        name = "Radio Test",
        description = "simply the best.",
        genre = "Mix",
        fallible = true,
        public = true,
        radio
)``

well it works now. I had to add fallible = true, now the playlist works. But would be interesting nevertheless where the error is. Installing a new version is not my number one choice as I normally use the apt-get manager and I don't want to destroy anything on my system.

Thanks again for your quick answer -

**Reply by vitoyucepi** [reply]
Here is a very basic setup that is based on your script.
```ruby
security = single("1.mp3")
hauptplaylist = playlist("/home/user/playlist.m3u")

radio = fallback([hauptplaylist, security], track_sensitive=false)

output.icecast(
  %mp3,
  radio,
  host = "localhost",
  port = 8000,
  password = "top secret",
  mount = "stream",
  name = "Radio Test",
  description = "simply the best.",
  genre = "Mix"
)
```
I've added `single` so it can be used in the `fallback` function to make it unfallable.

**Reply by suwofis** [reply]
Thanks a lot :)
